Framework/Algorithms/src/PolarizationCorrections/HeliumAnalyserEfficiency.cpp

```diff
@@ -0,0 +1,409 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include "MantidAlgorithms/PolarizationCorrections/HeliumAnalyserEfficiency.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/HistogramValidator.h"
+#include "MantidAPI/ITableWorkspace.h"
+#include "MantidAPI/WorkspaceUnitValidator.h"
+#include "MantidAlgorithms/PolarizationCorrections/SpinStateValidator.h"
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/CompositeValidator.h"
+#include "MantidKernel/ListValidator.h"
+
+#include <algorithm>
+#include <boost/math/distributions/students_t.hpp>
+#include <vector>
+
+namespace Mantid::Algorithms {
+// Register the algorithm into the algorithm factory
+DECLARE_ALGORITHM(HeliumAnalyserEfficiency)
+
+using namespace Kernel;
+using namespace API;
+
+namespace PropertyNames {
+static const std::string INPUT_WORKSPACE = "InputWorkspace";
+static const std::string OUTPUT_WORKSPACE = "OutputWorkspace";
+static const std::string P_HE = "HeliumPolarisation";
+static const std::string OUTPUT_T_WORKSPACE = "OutputTransmissionWorkspace";
+static const std::string OUTPUT_T_PARA_WORKSPACE = "OutputTransmissionParaWorkspace";
+static const std::string OUTPUT_T_ANTI_WORKSPACE = "OutputTransmissionAntiWorkspace";
+static const std::string SPIN_STATES = "SpinStates";
+static const std::string T_E = "TransmissionEmptyCell";
+static const std::string PXD = "GasPressureTimesCellLength";
+static const std::string COVARIANCE = "Covariance";
+static const std::string START_LAMBDA = "StartLambda";
+static const std::string END_LAMBDA = "EndLambda";
+static const std::string STOP_ON_FIT_ERROR = "StopOnFitError";
+} // namespace PropertyNames
+
+void HeliumAnalyserEfficiency::init() {
+  // Declare required input parameters for algorithm and do some validation here
+  auto validator = std::make_shared<CompositeValidator>();
+  validator->add<WorkspaceUnitValidator>("Wavelength");
+  validator->add<HistogramValidator>();
+  declareProperty(
+      std::make_unique<WorkspaceProperty<>>(PropertyNames::INPUT_WORKSPACE, "", Direction::Input, validator));
+  declareProperty(std::make_unique<WorkspaceProperty<>>(PropertyNames::OUTPUT_WORKSPACE, "PA", Direction::Output));
```

Comment:
I don't think we would normally provide a default for the output workspace, so I'd recommend that we just leave the default as an empty string here too. I found it a bit confusing when testing, so it would be good to be consistent with other algorithms.

Reply:
Done